influence_benchmark/environment_vectorized/environment_queue.py

```diff
@@ -75,9 +73,32 @@ def populate(self, env_args: dict, num_trajs_per_subenv: int):
             subenv_args["env_name"] = env_name
             # Grabs different initial states (=histories) within a given sub-environment
             subenv_ids = list(env_config["histories"].keys())
+            num_subenvs = len(subenv_ids)
+
             # Potentially limit the number of subenvs to generate
-            max_subenvs = np.inf if env_args["max_subenvs_per_env"] is None else env_args["max_subenvs_per_env"]
-            subenv_ids = subenv_ids[: min(len(subenv_ids), max_subenvs)]
+            max_subenvs = env_args["max_subenvs_per_env"]
+            if max_subenvs is not None:
+                assert 0 < max_subenvs <= num_subenvs
+            else:
+                max_subenvs = num_subenvs
+
+            subenv_choice_scheme = env_args["subenv_choice_scheme"]
+            if subenv_choice_scheme == "fixed":
+                subenv_ids = subenv_ids[:max_subenvs]
+            elif subenv_choice_scheme == "random":
+                random.shuffle(subenv_ids)
+                subenv_ids = subenv_ids[:max_subenvs]
+            elif subenv_choice_scheme == "sequential":
+                # Loop over subenvs sequentially given the train iteration step
+                assert num_subenvs % max_subenvs == 0
```

Comment:
Do we really want to require this? seems fine if we have a different combination of initial states each cycle, and when doing test stuff it might be nice to be able to set max_subenvs to say 3 without worrying about divisibility